lib/rules/no-for-in-array.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @fileoverview Disallow iterating over an array with a for-in loop
+ * @author Benjamin Lichtman
+ */
+"use strict";
+const ts = require("typescript");
+const util = require("../util");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * @type {import("eslint").Rule.RuleModule}
+ */
+module.exports = {
+    meta: {
+        docs: {
+            description: "Disallow iterating over an array with a for-in loop",
+            category: "Functionality",
+            recommended: false,
+            extraDescription: [util.tslintRule("no-for-in-array")],
+            url: util.metaDocsUrl("no-for-in-array"),
+        },
+        fixable: null,
+        schema: [],
+        type: "problem",
+    },
+
+    create(context) {
+        return {
+            ForInStatement(node) {
+                if (
+                    !context.parserServices ||
+                    !context.parserServices.program
+                ) {
+                    return;
+                }
```

Comment:
Does `context.parserServices.program` exist when `create` is first called, as implied by #230? If so, could you...

* ...return an empty object instead of one with `ForInStatement`?
* ...move the `checker = context...getTypechecker()` call to outside this returned function?

The two would be nice as precedent for other rules that would have multiple methods using the type checker.

Apologies if this is well known; I'm very new to ESLint. 😊

Reply:
you should use https://github.com/bradzacher/eslint-plugin-typescript/blob/master/lib/util.js#L114 

you need access to `context` and context is parameter of function `create(context)`